scripts/support/zephyr_flash_debug.py

```diff
@@ -38,105 +40,236 @@ def get_env_bool_or(env_var, default_value):
         return default_value
 
 
+def get_env_strip_or(env_var, to_strip, default_value):
+    value = os.environ.get(env_var, None)
+    if value is not None:
+        return value.strip(to_strip)
+    else:
+        return default_value
+
+
+def quote_sh_list(cmd):
+    '''Transform a command from list into shell string form.'''
+    fmt = ' '.join('{}' for _ in cmd)
+    args = [shlex.quote(s) for s in cmd]
+    return fmt.format(*args)
+
+
 def check_call(cmd, debug):
     if debug:
-        print(' '.join(cmd))
+        print(quote_sh_list(cmd))
     subprocess.check_call(cmd)
 
 
 def check_output(cmd, debug):
     if debug:
-        print(' '.join(cmd))
+        print(quote_sh_list(cmd))
     return subprocess.check_output(cmd)
```

Comment:
Are we doing the shell escaping ourselves because of Windows? Does it have anything similar to `execv()` that can be used instead?

Reply:
This uses the standard Python module shlex behind the scenes. In addition to making this portable for Windows, this also means that the output can be copy and pasted and the shell tokenization wil remain what the command in list form intended.

Reply:
Looking at this again, I think you might be asking about the subprocess.check_output line, not quote_sh_list(). If that's right, I'm not sure what you mean by "doing the shell escaping ourselves". The calling conventions for subprocess routines recommend using a sequence for reasons explained here:

https://docs.python.org/3/library/subprocess.html#frequently-used-arguments

Given that cmd is a list, this looks very similar to execv(), so I think it's doing what you're looking for. Please LMK if not.